src/momentintime.jl

```diff
@@ -39,14 +75,14 @@ abstract type YPFrequency{N} <: Frequency end
 
 A concrete frequency defined as 1 period per year.
 """
-struct Yearly <: YPFrequency{1} end
+struct Yearly{N} <: YPFrequency{1} where N<:Integer  end
```

Comment:
I think `where N<:Integer` here doesn't do anything and I find it confusing. Let's get rid of it.
Also, reusing `N` as the name of the type parameter can be misleading, because the `N` in `Yearly{N}` is not the same as the the `N` in `YPFrequency{N}`. Let's use a different name here. Maybe `P` for which _P_eriod of the year is the reference period ? I'm not sure ...

Reply:
I like the idea of improving the naming. It's `end_day` for Weekly and `end_month` for Quarterly and Yearly. I have pushed changes to reflect this.